dashboard/app/models/section.rb

```diff
@@ -77,8 +77,8 @@ def name_safe_students
     # normalization, this is necessary to ensure that we can recognize
     # the name on the other side
     self.students.each do |student|
-      normalized = student.name.split(/\s+/).join(' ')
-      trie.add normalized
+      student.name = student.name.strip.split(/\s+/).join(' ')
```

Comment:
Do we want to be mutating the student name? Note that if `save` is (accidentally) called elsewhere, we'll persist this change to the DB.